synapse/http/matrixfederationclient.py

```diff
@@ -739,8 +739,7 @@ async def post_json(
         )
         return body
 
-    @defer.inlineCallbacks
-    def get_json(
+    async def get_json(
```

Comment:
various things in TransportLayerClient that call this (`backfill`, `get_event`) have docstrings that claim to return Deferreds, which will no longer be true. It'd be nice to clean them up.

Reply:
Hmm...I thought I caught all those! I'll check again.

Reply:
I found a few more of these, good call! 👍 Pushed an additional fix.